data_model/src/events/pipeline.rs

```diff
@@ -93,19 +100,19 @@ pub enum EntityType {
     Transaction,
 }
 
-/// Entity type to filter events.
+/// [`Event`].
 #[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, IntoSchema)]
 pub struct Event {
-    /// Type of entity that caused this event.
+    /// [`EntityType`] of the entity that caused this [`Event`].
     pub entity_type: EntityType,
-    /// The status of this entity.
+    /// [`Status`] of the entity that caused this [`Event`].
     pub status: Status,
-    /// The hash of this entity.
+    /// [`Hash`] of the entity that caused this [`Event`].
     pub hash: Hash,
 }
 
 impl Event {
-    /// Constructs pipeline event.
+    /// Construct [`Event`].
```

Comment:
Why have this constructor? The `struct` literal syntax is going to be less verbose and more explicit.

Reply:
```rust
PipelineEvent::new(
    PipelineEntityType::Transaction,
    PipelineStatus::Validating,
    transaction.hash().into(),
)
```

```rust
PipelineEvent {
    entity_type: PipelineEntityType::Transaction,
    status: PipelineStatus::Validating,
    hash: transaction.hash().into(),
}
```
I don't think so